src/main/kotlin/org/rust/lang/core/psi/util/RustPsiNavigationExtensions.kt

```diff
@@ -146,12 +146,12 @@ private fun PsiElement.getTopmostParentInside(parent: PsiElement): PsiElement {
 
 private fun collectElements(start: PsiElement, stop: PsiElement?, pred: (PsiElement) -> Boolean): Array<out PsiElement> {
     val list = mutableListOf<PsiElement>()
-    var current = start
+    var current: PsiElement? = start
     while (current != stop) {
-        if (pred(current)) {
-            list.add(current)
+        if (pred(checkNotNull(current))) {
+            list.add(checkNotNull(current))
         }
-        current = current.nextSibling
+        current = checkNotNull(current).nextSibling
```

Comment:
I think a simple solution would be to add `current = current.nextSibling ?: break`. Also, I have a suspicion that this function can be neatly rewritten with the help of `generateSequence` (https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/generate-sequence.html).